tests/test_export_onnx.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import io
+import os
+import unittest
+import torch
+
+from detectron2 import model_zoo
+from detectron2.config import get_cfg
+from detectron2.export import add_export_config
+from detectron2.export.flatten import TracingAdapter
+from detectron2.modeling import build_model
+from detectron2.utils.env import TORCH_VERSION
+from detectron2.utils.testing import (
+    get_sample_coco_image,
+    SLOW_PUBLIC_CPU_TEST,
+    is_onnx_installed)
+
+
+SUPPORTED_ONNX_OPSET = 14
+
+
+# TODO: this test requires manifold access, see: T88318502
+class TestONNXTracingExport(unittest.TestCase):
+    def setUp(self):
+        if not is_onnx_installed():
```

Comment:
can you decorate with `skipIfNoONNX` instead?

Reply:
At first I thought about that too, but preferred the current approach, which is also used in other [tests](https://github.com/facebookresearch/detectron2/blob/main/tests/layers/test_mask_ops.py#L64)

The current approach generates a summary listing all tests that are disabled and including the reason, as shown below. The user is allowed to take an action from this.

```
pytest -sv tests/test_export_onnx.py 
==================================================================================================================================== test session starts ====================================================================================================================================
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0 -- /bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/detectron2/.hypothesis/examples')
rootdir: /detectron2
plugins: cov-3.0.0, hypothesis-6.44.0, typeguard-2.13.3, nbval-0.9.6, hydra-core-1.2.0.dev2
collected 5 items                                                                                                                                                                                                                                                                           

tests/test_export_onnx.py::TestONNXTracingExport::testCascadeRCNN SKIPPED (ONNX is not installed)
tests/test_export_onnx.py::TestONNXTracingExport::testMaskRCNNC4 SKIPPED (ONNX is not installed)
tests/test_export_onnx.py::TestONNXTracingExport::testMaskRCNNFPN SKIPPED (ONNX is not installed)
tests/test_export_onnx.py::TestONNXTracingExport::testMaskRCNNFPN_batched SKIPPED (ONNX is not installed)
tests/test_export_onnx.py::TestONNXTracingExport::testRetinaNet SKIPPED (ONNX is not installed)
```

Annotating `@skipIfNoONNX`, on the other hand, shows no warning. Users would never know about the existence of `tests/test_export_onnx.py` and the need of onnx for additional tests. See below

```
pytest -sv tests/test_export_onnx.py 
==================================================================================================================================== test session starts ====================================================================================================================================
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0 -- /bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/detectron2/.hypothesis/examples')
rootdir: /detectron2
plugins: cov-3.0.0, hypothesis-6.44.0, typeguard-2.13.3, nbval-0.9.6, hydra-core-1.2.0.dev2
collected 0 items
```

Annotating `@skipIfNoONNX` in each test individually has the same behavior as the `setUp` one, but it is tedious to duplicate it everywhere